AmplifyPlugins/API/AWSAPICategoryPlugin/Interceptor/SubscriptionInterceptor/IAMAuthInterceptor.swift

```diff
@@ -94,61 +113,82 @@ class IAMAuthInterceptor: AuthInterceptor {
-                                                 date: date,
-                                                 accept: RealtimeProviderConstants.iamAccept,
-                                                 contentEncoding: RealtimeProviderConstants.iamEncoding,
-                                                 contentType: RealtimeProviderConstants.iamConentType)
-        return authHeader
+        let remainingHeaders = mutableRequest.allHTTPHeaderFields?.filter { (header) -> Bool in
+            return !Self.lowercasedHeaderKeys.contains(header.key.lowercased())
+        }
+
+        return IAMAuthenticationHeader(host: host,
+                                       authorization: authorization,
+                                       securityToken: securityToken,
+                                       amzDate: amzDate,
+                                       accept: RealtimeProviderConstants.iamAccept,
+                                       contentEncoding: RealtimeProviderConstants.iamEncoding,
+                                       contentType: RealtimeProviderConstants.iamConentType,
+                                       remainingHeaders: remainingHeaders)
     }
 }
 
 /// Authentication header for IAM based auth
-private class IAMAuthenticationHeader: AuthenticationHeader {
+class IAMAuthenticationHeader: AuthenticationHeader {
     let authorization: String
     let securityToken: String
-    let date: String
+    let amzDate: String
     let accept: String
     let contentEncoding: String
     let contentType: String
+    let remainingHeaders: [String: String]?
 
-    init(authorization: String,
-         host: String,
-         token: String,
-         date: String,
+    init(host: String,
+         authorization: String,
+         securityToken: String,
+         amzDate: String,
          accept: String,
          contentEncoding: String,
-         contentType: String) {
-        self.date = date
+         contentType: String,
+         remainingHeaders: [String: String]?) {
         self.authorization = authorization
-        self.securityToken = token
+        self.securityToken = securityToken
+        self.amzDate = amzDate
         self.accept = accept
         self.contentEncoding = contentEncoding
         self.contentType = contentType
+        self.remainingHeaders = remainingHeaders
         super.init(host: host)
     }
 
-    private enum CodingKeys: String, CodingKey {
-        case authorization = "Authorization"
-        case accept
-        case contentEncoding = "content-encoding"
-        case contentType = "content-type"
-        case date = "x-amz-date"
-        case securityToken = "x-amz-security-token"
+    private struct DynamicCodingKeys: CodingKey {
+        var stringValue: String
+        init?(stringValue: String) {
+            self.stringValue = stringValue
+        }
+        var intValue: Int?
+        init?(intValue: Int) {
+            return nil
```

Comment:
Adding comment in code
```
// We are not using this, thus just return nil. If we don't return nil, then it is expected all of the
// stored properties are initialized, forcing the implementation to have logic that maintains the two
// properties `stringValue` and `intValue`. Since we don't have a string representation of an int value
// and aren't using int values for determining the coding key, then simply return nil since the encoder
// will always pass in the header key string.
```
Returning nil seems to keep it more simple. As a reference, i learned about this from [here](https://swiftsenpai.com/swift/decode-dynamic-keys-json/)

Reply:
thanks!